GVRf/Sample/gvrcamera2renderscript/src/org/gearvrf/sample/gvrcamera2renderscript/Camera2RenderscriptManager.java

```diff
@@ -0,0 +1,102 @@
+package org.gearvrf.sample.gvrcamera2renderscript;
+
+import org.gearvrf.GVRContext;
+import org.gearvrf.GVRScript;
+import org.gearvrf.GVRActivity;
+import org.gearvrf.GVRExternalTexture;
+import org.gearvrf.GVRScene;
+import org.gearvrf.GVRSceneObject;
+import org.gearvrf.GVRMaterial.GVRShaderType;
+import android.graphics.SurfaceTexture;
+import android.hardware.camera2.CameraAccessException;
+import android.renderscript.RenderScript;
+import android.util.Size;
+import java.util.Arrays;
+import java.util.List;
+import android.view.Surface;
+
+public class Camera2RenderscriptManager extends GVRScript
+{
+	private GVRActivity mActivity;
+	private RenderScript mRS;
+	private GVRContext mGVRContext;
+	private SurfaceTexture mSurfaceTexture;
+	private SurfaceTexture mEffectTexture;
+	private Camera2Helper mCameraHelper;
+	private RenderscriptProcessor mProcessor;
+	private Surface mSurfaceInterim;
+	
+	public Camera2RenderscriptManager(GVRActivity activity)
+	{
+		mActivity = activity;
+		mRS = RenderScript.create(mActivity);
+	}
+	
+	@Override
+	public void onInit(GVRContext gvrContext) throws Throwable 
+	{
+		mGVRContext = gvrContext;
+		GVRScene mainScene = gvrContext.getNextMainScene(new Runnable() {
+			
+			@Override
+			public void run() 
+			{
+				try
+				{
+					mCameraHelper = new Camera2Helper(mActivity, 0);
+					Size previewSize = mCameraHelper.setPreferredSize(1920, 1080);
```

Comment:
Hi liaxim, yes, the frame-rate depends on the preview size. Actually. I'm using 1280x720 in my real developments for the performance issue. And also that's why I made setPreferredSize method. Someone who would use this code might need to find the most proper sizes in their use cases.